### ISSUE TITLE
Improve computation and type inference of the default value for each rule

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#808]
## AS IS

```ts
createRule<string | null>({
  defaultValue: undefined,
  async verify({ document }) {
    document.rule.value; // => Readonly<string | null>
  },
});
```

```ts
createRule<string | null>({
  // omitted `defaultValue`
  async verify({ document }) {
    document.rule.value; // => Readonly<string | null>
  },
});
```

The type of `document.rule.value` is `Readonly<string | null>`, but its actual value is `true`.

## TO BE

```ts
createRule<string | null>({
  defaultValue: undefined, // <= EXPECTED TYPE ERROR
  async verify({ document }) {
    ...
  },
});
```

```ts
createRule<string | null>({ // <= EXPECTED TYPE ERROR, Urge to specify `defaultValue`
  async verify({ document }) {
    ...
  },
});
```

```ts
createRule<boolean>({ // <= No error, `defaultValue` will be `true`
  async verify({ document }) {
    ...
  },
});
```

```ts
createRule({ // <= No error, `defaultValue` will be `true`
  async verify({ document }) {
    ...
  },
});
```

I've identified a problem with the `createRule` function where there is no type error when the `defaultValue` is omitted or specified as `undefined`. I would like to improve this by causing a type error when the `defaultValue` is not specified or set to undefined. Additionally, I want to address the issue of the default `defaultValue` being set to true.
